File: AUGUST-2025/12-08-2025/p2787.cs
public class Solution {
    public int NumberOfWays(int n, int x) {
        List<int> powers = new List<int>();
          int i = 1;

          while(true){
               int val =1;

               for(int j =0; j < x;j++){
                   val *=i;

                    if(val > n) break;
               }

            if(val > n) break;

           powers.Add(val);
           i++;
          }
        
      int MOD = 1_000_000_007;
     int[] dp = new int[n+1];
     dp[0] = 1;

    foreach(int power in powers){
        for(int j = n;j >= power;j--){
            dp[j] = (dp[j] + dp[j-power]) % MOD;
        }
    }
   return dp[n];

    }
}

File: AUGUST-2025/12-08-2025/p2787.php
class Solution {

    /**
     * @param Integer $n
     * @param Integer $x
     * @return Integer
     */
    function numberOfWays($n, $x) {
        $MOD = 1000000007;

     $dp = array_fill(0, $n + 1, 0);
    $dp[0] = 1;

     for ($i = 1; ; $i++) {
        $power = $this->pow_limited($i, $x, $n); 
        if ($power > $n) break;

         for ($s = $n; $s >= $power; $s--) {
            $dp[$s] = ($dp[$s] + $dp[$s - $power]) % $MOD;
        }
    }

    return $dp[$n];
    }

    function pow_limited(  $base,   $exp,   $limit) {
    $result = 1;
    for ($e = 0; $e < $exp; $e++) { 
        if ($base !== 0 && $result > intdiv($limit, $base)) {
            return $limit + 1; 
        }
        $result *= $base;
        if ($result > $limit) return $limit + 1;
    }
    return $result;
}
}